plugins/woocommerce-admin/client/core-profiler/pages/Loader.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { useState, useEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { CoreProfilerStateMachineContext } from '..';
+import ProgressBar from '../components/progress-bar/progress-bar';
+import { getLoaderStageMeta } from '../get-loader-stage-meta';
+
+export type Stage = {
+	title: string;
+	image?: string | JSX.Element;
+	paragraphs: Array< {
+		label: string;
+		text: string;
+		duration?: number;
+	} >;
+};
+
+export type Stages = Array< Stage >;
+
+export const Loader = ( {
+	context,
+}: {
+	context: CoreProfilerStateMachineContext;
+} ) => {
+	const stages = getLoaderStageMeta( context.loader.useStages ?? 'default' );
+	const currentStage = stages[ context.loader.stageIndex ?? 0 ];
+	const [ currentParagraph, setCurrentParagraph ] = useState( 0 );
+
+	useEffect( () => {
+		const interval = setInterval( () => {
+			currentStage.paragraphs[ currentParagraph + 1 ]
```

Comment:
The linting is failing. 

```
36:4  error  Expected an assignment or function call and instead saw an expression           no-unused-expressions
38:9  error  'currentParagraph' is already declared in the upper scope on line 32 column 10  @typescript-eslint/no-shadow
```

I think maybe we can rewrite it like below:

```tsx
			setCurrentParagraph( ( _currentParagraph ) =>
				currentStage.paragraphs[ _currentParagraph + 1 ]
					? _currentParagraph + 1
					: 0
			);
```

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/38019/commits/ec092926082634b5ee0b60c0ca9fe7f274af52e4